BaseClasses.py

```diff
@@ -487,8 +487,10 @@ def get_placeable_locations(self, state=None, player=None) -> List[Location]:
     def get_unfilled_locations_for_players(self, location_names: List[str], players: Iterable[int]):
         for player in players:
             if not location_names:
-                location_names = [location.name for location in self.get_unfilled_locations(player)]
-            for location_name in location_names:
+                valid_locations = [location.name for location in self.get_unfilled_locations(player)]
+            else:
+                valid_locations = location_names.copy()
```

Comment:
I don't think there is a need to copy. 
```suggestion
                valid_locations = location_names
```

Reply:
Since we don't change valid_locations afterwards, I agree.